community/bolt/src/main/java/org/neo4j/bolt/v1/runtime/TransactionStateMachine.java

```diff
@@ -196,7 +199,16 @@ State run( MutableTransactionState ctx, SPI spi, String statement,
                     {
                         if ( BEGIN.matcher( statement ).matches() )
                         {
-                            ctx.currentTransaction = spi.beginTransaction( ctx.securityContext );
+                            AnyValue txTimeoutMills = params.get( TX_TIMEOUT_MILLIS );
+                            if ( txTimeoutMills instanceof LongValue )
+                            {
+                                ctx.currentTransaction = spi.beginTransaction( ctx.securityContext,
+                                        ((LongValue) txTimeoutMills).value(), TimeUnit.MILLISECONDS );
+                            }
+                            else
+                            {
+                                ctx.currentTransaction = spi.beginTransaction( ctx.securityContext );
```

Comment:
I think we should only go here if `txTimeoutMills == Values.NO_VALUE`. Exception should be thrown if it's not a `NO_VALUE` and not an instance of `NumberValue`